### PR TITLE
add efficient user context manipulation routines

### PR DESCRIPTION
**Pull request by PetrShumilov**
Goals:
* Remove useless `rt_sigprocmask` syscall invocation during user context manipulation 
* Add platform-independent API for user context-related things

Related issues: KPHP-1944

### CODE REVIEW

File: server/php-runner.cpp [hunk 1]
```diff
@@ -80,13 +81,24 @@ void send_slow_net_event_stats(const net_event_t &event, double time_sec) noexce
...
   assert (in_script_context == true);
   in_script_context = false;
   current_script->state = run_state_t::error;
   current_script->error_message = error_message;
   current_script->error_type = error_type;
-  stack_end = reinterpret_cast<char *>(exit_context.uc_stack.ss_sp) + exit_context.uc_stack.ss_size;
+  stack_end = reinterpret_cast<char *>(get_context_stack_ptr_portable(exit_context)) + get_context_stack_size_portable(exit_context);
+
+#if defined(__linux__) && defined(__x86_64__)
+  // The PhpScript::error may be produced in process of signal handling. The default behavior on Linux-based platforms
```
Comment (astrophysik) [inline server/php-runner.cpp:93]: We also have own implementation for `darwin aarch64`. Do we need same logic for this?

File: server/php-runner.cpp [hunk 2]
```diff
@@ -477,7 +489,7 @@ void PhpScript::run() noexcept {
       php_assert(CurException.is_null());
       CurException = saved_exception;
     }
-    error(php_uncaught_exception_error(CurException), script_error_t::exception);
+    error(php_uncaught_exception_error(CurException), script_error_t::exception, /* no triggered by signal */ std::nullopt);
```
[the remark this answers is not in the document]
Comment (PetrShumilov) [reply server/php-runner.cpp:492]: I don't think so. It's a good hint for reader, furthermore for developers without IDE. I suppose it's best practice. 
Reader first!!!

File: server/php-runner.cpp (hunk 1 above)
Comment (PetrShumilov) [reply server/php-runner.cpp:93]: I will check it. Previously, we also had own implementation for `darwin aarch64` but no needed a special handling for this case. I may be a result of different signal subsystems.

Comment (PetrShumilov) [reply server/php-runner.cpp:93]: Checked, thanks for notice, fixed for `darwin aarch64`

**Review by DrDet — APPROVED** [review]

**Review by DrDet — APPROVED** [review]

**Pull request merged**